bazel/envoy_test.bzl

```diff
@@ -109,18 +110,13 @@ def envoy_cc_fuzz_test(
         **kwargs
     )
     cc_test(
```

Comment:
Can we make this a `cc_binary( testonly = True )` instead? cc @asraa

Reply:
We should make this a _binary. Otherwise running bazel test //... will attempt to run the <name>_binary's